examples/shapes/sphereCotangentLaplaceOperator.cpp

```diff
@@ -0,0 +1,336 @@
+/**
+  * This program is free software : you can redistribuate it and/or modify
+  * it under the terms of the GNU Lesser General Public License as
+  * published by the Free Software Foundation, either version 3 of the
+  * License, or (at you option) any later version.
+  *
+  * This program is distributed int the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  * GNU General Public License for more details.
+  *
+  * You should have received a copy of the GNU General Public License
+  * along with this program. If not, set <http://www.gnu.org/license/>.
+**/
+
+/**
+ * @file shapes/exampleTriangulatedSurface.cpp
+ * @ingroup Examples
+ * @author Thomas Caissard (\c thomas.caissard@liris.cnrs.fr )
+ * Laboratoire d'InfoRmatique en Image et Systèmes d'information - LIRIS (CNRS, UMR 5205), INSA-Lyon, France
+ * LAboratoire de MAthématiques - LAMA (CNRS, UMR 5127), Université de Savoie, France
+ *
+ * @date 2017/06/20
+ *
+ * An example file named exampleTriangulatedSurface.
+ *
+ * This file is part of the DGtal library.
+**/
+
+//! [TriangulatedSurfaceUsageFull]
+
+#define WITH_3D
+
+////////// INCLUDES /////////
+
+#include <DGtal/helpers/StdDefs.h>
+
+#include <DGtal/base/BasicFunctors.h>
+
+#include <DGtal/topology/DigitalSurface.h>
+#include <DGtal/topology/SetOfSurfels.h>
+#include <DGtal/topology/LightImplicitDigitalSurface.h>
+
+#include <DGtal/io/viewers/Viewer3D.h>
+#include <DGtal/io/colormaps/ColorBrightnessColorMap.h>
+
+#include <DGtal/geometry/surfaces/estimation/LocalEstimatorFromSurfelFunctorAdapter.h>
+#include <DGtal/geometry/surfaces/estimation/IIGeometricFunctors.h>
+#include <DGtal/geometry/surfaces/estimation/IntegralInvariantCovarianceEstimator.h>
+
+#include <DGtal/shapes/parametric/Ball3D.h>
+#include <DGtal/shapes/Shapes.h>
+#include <DGtal/shapes/GaussDigitizer.h>
+#include <DGtal/shapes/Mesh.h>
+#include <DGtal/shapes/MeshHelpers.h>
+#include "DGtal/shapes/TriangulatedSurface.h"
+
+#include <DGtal/math/linalg/EigenSupport.h>
+
+///////// NAMESPACES //////////
+
+using namespace std;
+using namespace DGtal;
+using namespace Eigen;
+using namespace Z3i;
+
+////////// TYPEDEFS //////////
+
+typedef Z3i::Space Space;
+typedef Z3i::KSpace KSpace;
+typedef Z3i::RealVector RealVector;
+typedef Z3i::RealPoint RealPoint;
+typedef Z3i::Point Point;
+
+RealPoint cartesian_to_spherical(const RealPoint & a)
+{
+  return RealPoint(a.norm(), atan2(a[1], a[0]), acos(a[2]));
+}
+
+struct Options
+{
+  double h;
+  int function;
+  int smooth;
+
+  std::string error_output;
+};
+
+template <typename Shape>
+void laplacian(Shape& shape, const Options& options,
```

Comment:
please add some doc, even for examples.
(maybe not as dense as technical doc of classes but still)